### ISSUE TITLE
_error for objects in FieldArray

### ISSUE THREAD

**Issue by halayli** [erikras/redux-form#965]
Sometimes non-field errors can be at the object level inside a FieldArray. It would be useful to support _error for those objects as well and when any of the values in the object change it clears out.

Thoughts?

**Comment by erikras** [comment]
Can you give me a concrete example?

**Comment by halayli** [comment]
Yes definitely.

let's say you have the following structure:

```
{
  "id": 1234,
  "contacts": [
    {
      "id": 123,
      "email": <email>,
      "phone", <phone>
    },...
  ]
}
```

I have a requirement that either email or phone should be filled out. The error is not email/phone field specific but contact specific. Showing the error near email or phone would be confusing also if let's say I put the error on 'email' and the user filled out the phone field the form will not submit because it will still be in an invalid state.

**Comment by erikras** [comment]
Ah!

> ***at the object level***

!! Got it.

One thing about running a library is that you can be certain that there is no end of reasonable suggestions that you have not thought of from users. I'll mark this as an enhancement and see what I can do.

**Comment by halayli** [comment]
Agreed! especially when it comes to forms and trying to accommodate the various patterns they can have.

I'll try to take a stab at it on the weekend as well.

**Comment by erikras** [comment]
@halayli I went to test this, and it really looks like it's working (as far as I am understanding your requirements). [Check this test out](https://github.com/erikras/redux-form/blob/114d9c915ec7b7f3d0283ff122d9f888af3ec08a/src/__tests__/FieldArray.spec.js#L214-L252). Is that not what you mean?

If it's sync validation, it should be "cleared out" when you make the value pass validation.

**Comment by halayli** [comment]
@erikras shouldn't the FieldArray name be "foo" instead of "foo[0]"?

**Comment by halayli** [comment]
http://redux-form.com/6.0.0-alpha.8/examples/fieldArrays/

Basically I am trying to do the equivalent of setting an _error on `const memberErrors = {}` in the validate function.

**Comment by erikras** [comment]
@halayli Yes, you are totally correct. I was wrong.

**Comment by erikras** [comment]
Okay, there was a tiny bug here. The way you do object level errors for array objects is [like this](https://github.com/erikras/redux-form/blob/v6/src/__tests__/FieldArray.spec.js#L232-L257). The fix will be in the next release.

**Comment by ooflorent** [comment]
@erikras Can we close this one now?

**Comment by erikras** [comment]
Sure. I'll leave a comment here when the release goes out.

**Comment by erikras** [comment]
Published as [`v6.0.0-alpha.11`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-alpha.11).
